### PR TITLE
SAAS-1149 Service Providers API

### PR DESCRIPTION
**Pull request by czarnecki-piotr**

### CODE REVIEW

File: docs/products/access/api-reference/config-api/service-providers-api.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+* Endpoint: `/api/v1/saml-sp-config`
+* Method: POST
+
+JSON body parameters:
+
+| Param                     | Required | Example                           | Description                                                                      |
+|---------------------------|----------|-----------------------------------|----------------------------------------------------------------------------------|
+| name                      | yes      | "Service Provider name"           | Service Provider name.                                                           |
+| metadata_type             | yes      | "XML"                             | Type of Service Provider metadata. <br/> Allowed values: `URL`, `XML`, `MANUAL`. |
+| metadata_xml              | depends  | Xml file content                  | Required when metadata type is Xml.                                              |
```
Comment (karolrom-thales) [inline docs/products/access/api-reference/config-api/service-providers-api.md:62]: Xml -> XML

File: docs/products/access/api-reference/config-api/service-providers-api.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+| metadata_url              | depends  | "https://example.sp.com/metadata" | Location of the remote Xml metadata. Required when metadata type is URL.         |
+| manual_metadata           | depends  | See table below                   | Metadata in form of Json object. Required when metadata type is MANUAL.          |
```
Comment (karolrom-thales) [inline docs/products/access/api-reference/config-api/service-providers-api.md:64]: Json -> JSON

File: docs/products/access/api-reference/config-api/service-providers-api.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+}
+```
+
+Example success response:
+
+```http
+HTTP/1.1 201 CREATED
+Content-Type: application/json;charset=UTF-8
+Cache-Control: no-store
+Pragma: no-cache
```
Comment (karolrom-thales) [inline docs/products/access/api-reference/config-api/service-providers-api.md:127]: I think we should return Location header

File: docs/products/access/api-reference/config-api/service-providers-api.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
+# Service Providers API
+
+This allows the creation of new Service Provider via a REST API.
+It can be utilized in scripts to add, list, edit, or delete service providers.
+
+All endpoints are protected with API client. 
+It requires an API client with the `onegini_api_config` scope (Config API).
```
Comment (karolrom-thales) [inline docs/products/access/api-reference/config-api/service-providers-api.md:7]: I'm wondering if it's clear to everyone what it means to create a Service Provider. It's more about registering an existing Service Provider than creating one. What is being created is obviously a resource representing such a service provider, but not the service provider itself.

File: docs/products/access/api-reference/config-api/service-providers-api.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+### Create Service Provider
+
+This creates a new Service Provider
+
+* Endpoint: `/api/v1/saml-sp-config`
+* Method: POST
+
+JSON body parameters:
+
+| Param                     | Required | Example                           | Description                                                                      |
```
Comment (karolrom-thales) [inline docs/products/access/api-reference/config-api/service-providers-api.md:58]: As this table is common for all methods, maybe it should be defined before `POST`

Comment (czarnecki-piotr) [reply docs/products/access/api-reference/config-api/service-providers-api.md:58]: done

File: docs/products/access/api-reference/config-api/service-providers-api.md (hunk 2 above)
Comment (czarnecki-piotr) [reply docs/products/access/api-reference/config-api/service-providers-api.md:64]: done

File: docs/products/access/api-reference/config-api/service-providers-api.md (hunk 4 above)
Comment (czarnecki-piotr) [reply docs/products/access/api-reference/config-api/service-providers-api.md:7]: Good point. I changed it to create/update "Service Provider configuration"

File: docs/products/access/api-reference/config-api/service-providers-api.md (hunk 3 above)
Comment (czarnecki-piotr) [reply docs/products/access/api-reference/config-api/service-providers-api.md:130]: I created a PR for that: https://github.com/onewelcome/access/pull/2024

**Review by karolrom-thales — APPROVED** [review]

**Pull request merged**